src/templates/cmp.js

```diff
@@ -0,0 +1,175 @@
+      );
+      iframe.setAttribute('style', 'border:0;outline:0;width:0;height:0;');
+      iframe.setAttribute('frameborder', '0');
+
+      iframe.addEventListener('load', function () {
+        window.__tcfapi = function (command, version, callback, parameter) {
+          message('cmd;' + command + ';' + version + ';' + parameter, callback);
+        };
+
+        window.addEventListener(
+          'message',
+          function (ev) {
+            try {
+              if (ev.origin === '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>' && ev.data) {
+                var message = ev.data.split(';');
+                var position = message[0] === 'err' ? 1 : 0;
+                var id = message[position];
+                var callback = callbackMap[id][position];
+                if (logCallbackIndex + '' !== id) delete callbackMap[id];
+                var r = JSON.parse(atob(message[++position]));
+                var s = message[++position] ? true : undefined;
+                if (callback) {
+                  callback(r, s);
+                }
+              }
+            } catch (e) {}
+          },
+          false,
+        );
+
+        onAPILoaded('iframe');
+      });
+
+      iframe.addEventListener('error', log.bind(null, 'loaded', false, { type: 'iframe' }));
+
+      return iframe;
+    }
+
+    var iframeRetries = 0;
+
+    function loadIframe() {
+      if (iframeRetries >= 3) {
+        return;
+      }
+
+      if (!document.body) {
+        setTimeout(function () {
+          iframeRetries++;
+          loadIframe();
+        }, 100);
+        return;
+      }
+
+      iframe = createIframe();
+      document.body.appendChild(iframe);
+    }
+
+    function createTcfapiScriptTag() {
+      var xt = '<%-XT%>';
+      var hasConsent;
+
+      if (window.localStorage && localStorage.getItem && localStorage.setItem) {
+        if (!localStorage.getItem('<%-TECH_COOKIE_NAME%>')) {
+          // no tech info yet, need to store
+          localStorage.setItem('<%-TECH_COOKIE_NAME%>', xt);
+        } else {
+          xt = localStorage.getItem('<%-TECH_COOKIE_NAME%>'); // prefer tech info from localStorage
+          hasConsent = localStorage.getItem('<%-COOKIE_NAME%>');
+        }
+      }
+
+      var tcfapiScriptTag = document.createElement('script');
+      tcfapiScriptTag.setAttribute('type', 'text/javascript');
+      tcfapiScriptTag.setAttribute(
+        'src',
+        '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/tcfapi.js?<%-TECH_COOKIE_NAME%>=' +
+          xt +
+          (hasConsent !== null ? '&consent=' + hasConsent : '') +
+          (channelId !== '' ? '&channelId=' + channelId : ''),
+      );
+
+      tcfapiScriptTag.addEventListener('error', log.bind(null, 'loaded', false, { type: '3rdparty' }));
+      tcfapiScriptTag.addEventListener('load', function () {
+        // if not an Opera (Presto) browser, we load the iframe into the host document
+        if (
```

Comment:
With this condition here it loads the iframe-less API every time. Which could be fine if you remove 
```const tcfapiJs = await renderFile(path.join(__dirname, "../templates/tcfapi.js"), values);```
from `tcfapiIframe.ts`.